apps/client/src/app/structure-definition/structure-definition.component.ts

```diff
@@ -300,6 +302,22 @@ export class StructureDefinitionComponent extends BaseComponent implements OnIni
       }, (err) => {
         this.message = getErrorString(err);
       });
+
+    const implementationGuideId = this.route.snapshot.paramMap.get('implementationGuideId');
+    const implementationGuide = await this.implementationGuideService.getImplementationGuide(implementationGuideId).toPromise();
+    const resources = (<ImplementationGuide> implementationGuide).definition.resource;
+
+
+    const index = resources.findIndex(resource => {
+      return resource.reference.reference.indexOf(this.structureDefinition.id) > 0;
+    });
+    if(index === -1) throw new Error(`Couldn't find structure definition in resources`);
+
+    (<ImplementationGuide> implementationGuide).definition.resource[index].name =
+      this.structureDefinition.title ? this.structureDefinition.title : this.structureDefinition.name;
+
+    await this.implementationGuideService.saveImplementationGuide(<ImplementationGuide> implementationGuide)
+      .subscribe( () => {});
```

Comment:
Don't think `.susbcribe()` is needed here if you're not doing anything with it.
Might consider something like this instead:
```
await this.implementationGuideService.saveImplementationGuide(<ImplementationGuide> implementationGuide)
  .toPromise()
  .catch(err => console.log(err));
```

Reply:
Added the requested changes.  Let me know if you think anything else is needed